src/main/webapp/assets/js/feedScript.js

```diff
@@ -1237,24 +1237,30 @@ function limitCharacterInput(e) {
 }
 
 /**
- * Populates the post modal's time slots with 'smart' values
- * based on the current time.
+ * Populates the post modal's datetime slots with 'smart' values
+ * based on the current datetime.
  */
-function populatePostModalTime() {
+function populatePostModalDateTime() {
+  const modalMonth = document.getElementById('modal-month');
+  const modalDay = document.getElementById('modal-day');
   const modalStartHour = document.getElementById('modal-start-hour');
   const modalStartMinute = document.getElementById('modal-start-minute');
   const startAmOrPm = document.getElementById('start-am-or-pm');
   const modalEndHour = document.getElementById('modal-end-hour');
   const modalEndMinute = document.getElementById('modal-end-minute');
   const endAmOrPm = document.getElementById('end-am-or-pm');
 
+  modalMonth.value = new Date().getMonth() + 1; // Months are indexed at 0
+  modalDay.value = new Date().getDate();
+
   const nowHours = parseInt(new Date().getHours());
-  modalStartHour.value = nowHours > 12 ? nowHours - 12 : nowHours;
+  modalStartHour.value = nowHours > 12 ? nowHours - 12 : (nowHours !== 0 ? nowHours : 12);
```

Comment:
Explain what this does to make it clearer for anyone reading this with not much context.

Reply:
Will do!